Sources/Fuzzilli/Core/ProgramBuilder.swift

```diff
@@ -1170,15 +1170,17 @@ public class ProgramBuilder {
 
         currentCodegenBudget -= 1
 
+        // Update our analyses
+        scopeAnalyzer.analyze(instr)
+        contextAnalyzer.analyze(instr)
+
         // Update type information
         let typeChanges = interpreter?.execute(instr) ?? []
         for (variable, type) in typeChanges {
+            assert(scopeAnalyzer.visibleVariables.contains(variable))
```

Comment:
Can you also add an assert here that `type != typeOf(variable)`?

Reply:
This is a bit complicated because current API returns .unknown when type not set. So we cannot distinguish between (unknown -> unknown and nil->unknown change).
So do we want to extend API of InterpreterState and ProgramTypes just because this assert with:

maybeType(variable) -> Type?
type(variable) -> Type (unknown as default)
?

Reply:
Ah right, makes sense. How about `assert(type != typeOf(variable) || type == .unknown)` and a short comment explaining that we can't distinguish no type from unknown type here?